src/main/kotlin/no/nav/familie/ba/sak/kjerne/vedtak/vedtaksperiode/vedtakBegrunnelseProdusent/VilkårFilterUtil.kt

```diff
@@ -52,22 +52,21 @@ fun ISanityBegrunnelse.matcherMedUtdypendeVilkår(vilkårResultat: VilkårResult
     }
 }
 
-private fun Collection<UtdypendeVilkårsvurdering>.erLik(
-    utdypendeVilkårsvurderingFraSanityBegrunnelse: List<VilkårTrigger>?,
+private fun List<UtdypendeVilkårsvurdering>.erLik(
+    utdypendeVilkårsvurderingFraSanityBegrunnelse: List<VilkårTrigger>,
 ): Boolean {
-    val utdypendeVilkårPåVilkårResultat = this.toSet()
-    val utdypendeVilkårPåSanityBegrunnelse: Set<UtdypendeVilkårsvurdering> =
-        utdypendeVilkårsvurderingFraSanityBegrunnelse?.tilUtdypendeVilkårsvurderinger()?.toSet() ?: emptySet()
+    val altErLikt =
+        utdypendeVilkårsvurderingFraSanityBegrunnelse.all {
+            it.stemmerMedVilkårsvurdering(utdypendeVilkårPåVilkårResultat = this)
+        }
 
-    return utdypendeVilkårPåSanityBegrunnelse.isEmpty() || utdypendeVilkårPåVilkårResultat == utdypendeVilkårPåSanityBegrunnelse
+    return utdypendeVilkårsvurderingFraSanityBegrunnelse.isEmpty() || altErLikt
```

Comment:
`.all{}` på empty list blir alltid true

https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS45LjIzIiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiLyoqXG4gKiBZb3UgY2FuIGVkaXQsIHJ1biwgYW5kIHNoYXJlIHRoaXMgY29kZS5cbiAqIHBsYXkua290bGlubGFuZy5vcmdcbiAqL1xuXG5cbmZ1biBtYWluKCkge1xuICAgIHByaW50bG4oZW1wdHlMaXN0PFN0cmluZz4oKS5hbGx7IGZhbHNlIH0pXG59In0=

```suggestion
    return altErLikt
```